### ISSUE TITLE
Training settings

### ISSUE THREAD

**Issue by hkzhang-git** [wilile26811249/MobileViT#4]
I really appreciate your efforts in implementing this model in pytorch. Here, I have one concern about the training settings. If what I understand is correct, you just trained the model for less than 5 epoches. 

In addition, the hyper-parameters you adopted is  different from that in the original article. For instance, in the original manuscript, authors train mobilevit using AdamW optimizer, label smoothing cross-entry and multi-scale sampler. The training phase has a warmup stage. 

I also found that the classificaion accuracy provided here is much lower than that in the original version. 

I conjecture that the gab between accuracies are caused by different training settings.

**Comment by wilile26811249** [comment]
Yes, the accuracy gap is caused by the different training setting.
Here, I'm focus on the network architecture first.
The training settings are written by me before, and I have been using it until now.
Maybe, I should try use timm for the training setting to improve the training accuracy.

Thank you for your suggestion!

**Comment by hkzhang-git** [comment]
Thanks for your reply, I'm trying to reproduce the accuracy listed in their paper. After read your implementation, I have another issue that is the head num for attention module is set to 1 for every Transformer module. How did you know that head num should be set to 1. I can not find any discription about this in the original article.

**Comment by wilile26811249** [comment]
Yes, the authors were not mentioned about the number of the attention heads in the paper.
So I didn't set a certain number about the attention head in particular.
But if you follow other papers and preferences, setting the heads number to 8 should be a better choice.

**Comment by hkzhang-git** [comment]
Thanks for your replies, which solve all my concerns.

**Comment by hkzhang-git** [comment]
Hi buddy, the official code is available at https://github.com/apple/ml-cvnets. I think you may be interested in this.

**Comment by wilile26811249** [comment]
Wow, that's great.
Thank you for your information.
